ansible_ai_connect/ai/api/tests/test_views.py

```diff
@@ -195,12 +198,17 @@ def setUp(self):
         super().setUp()
         self.username = "u" + "".join(random.choices(string.digits, k=5))
         self.password = "secret"
-        email = "user@example.com"
+        self.email = "user@example.com"
+        if hasattr(self, "_inhibit_create_user"):
+            return
+        logger.warn("Please move this test to WisdomServiceAPITestCaseBaseOIDC")
+
         self.user = get_user_model().objects.create_user(
```

Comment:
I was thinking.. and hence my delay in re-reviewing.

Would it not be better to have something like:
```
def setUp(self):
    super().setUp()
    self.username = "u" + "".join(random.choices(string.digits, k=5))
    self.password = "secret"
    self.email = "user@example.com"
    self.user = create_user() # <--- Defer to a method that can be overridden

    self.user.user_id = str(uuid.uuid4())
    self.user.community_terms_accepted = timezone.now()
    self.user.save()

    group_1, _ = Group.objects.get_or_create(name="Group 1")
    group_2, _ = Group.objects.get_or_create(name="Group 2")
    group_1.user_set.add(self.user)
    group_2.user_set.add(self.user)
    cache.clear()
```
where the implementation in `WisdomServiceAPITestCaseBase` is:
```
def create_user(self):
    logger.warn("Please move this test to WisdomServiceAPITestCaseBaseOIDC")
    return get_user_model().objects.create_user(
        username=self.username,
        email=self.email,
        password=self.password,
    )
```
Then override the method in `WisdomServiceAPITestCaseBaseOIDC` with:
```
def create_user(self):
    return create_user_with_provider(
        username=self.username,
        email=self.email,
        password=self.password,
        provider=USER_SOCIAL_AUTH_PROVIDER_OIDC,
        rh_org_id=1981,
        social_auth_extra_data={},
    )
```

Reply:
Yes, this is indeed way more elegant.